### PR TITLE
xtrx: fix PCI DMA allow/free with kernel >= 5.18

### PR DESCRIPTION
**Pull request by trabucayre**
Since kernel **5.18.0** `pci_alloc_consistent` `pci_free_consistent` and `pci_set_consistent_dma_mask` are no more available and must be replaced by `dma_alloc_coherent` `dma_free_coherent` and `dma_set_mask_and_coherent`
This PR add preprocessing test using kernel version to keep backward compatibility and fix for kernel >= 5.18.

### CODE REVIEW

File: xtrx.c [hunk 1]
```diff
@@ -27,6 +27,9 @@
 #include <linux/kdev_t.h>
 #include <linux/fs.h>
 #include <linux/cdev.h>
+#ifdef CONFIG_CPU_RK3399
```
Comment (chemeris) [inline xtrx.c]: Is this related to the patch or an accidental?

If it's related - please explain the reasoning in the commit message, or split into separate commits.

Comment (trabucayre) [reply xtrx.c]: it's accidental (but without this fix driver won't build on rockpro and CM4)

**Comment by trabucayre** [comment]
Fixed

File: xtrx.c (hunk 1 above)
Comment (chemeris) [reply xtrx.c]: Thank you. Please submit the CONFIG_CPU_RK3399 fix separately.
